packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -570,35 +571,35 @@ async function prepareResource(context: $TSContext, resource: $TSAny) {
 
   const cfnFile = cfnFiles[0];
   const cfnFilePath = path.normalize(path.join(resourceDir, cfnFile));
-  const cfnMeta = JSONUtilities.readJson<$TSAny>(cfnFilePath);
+  const { cfnTemplate } = await readCFNTemplate(cfnFilePath);
   const paramType = { Type: 'String' };
 
   if (resource.service === FunctionServiceNameLambdaLayer) {
-    cfnMeta.Parameters.deploymentBucketName = paramType;
-    cfnMeta.Parameters.s3Key = paramType;
+    cfnTemplate.Parameters.deploymentBucketName = paramType;
+    cfnTemplate.Parameters.s3Key = paramType;
     storeS3BucketInfo(category, s3Bucket, envName, resourceName, s3Key);
   } else if (resource.service === ApiServiceNameElasticContainer) {
     const cfnParams = { ParamZipPath: s3Key };
 
     const cfnParamsFilePath = path.normalize(path.join(resourceDir, 'parameters.json'));
     JSONUtilities.writeJson(cfnParamsFilePath, cfnParams);
   } else {
-    cfnMeta.Parameters.deploymentBucketName = paramType;
-    cfnMeta.Parameters.s3Key = paramType;
-    if (cfnMeta.Resources.LambdaFunction.Type === 'AWS::Serverless::Function') {
-      cfnMeta.Resources.LambdaFunction.Properties.CodeUri = {
+    cfnTemplate.Parameters.deploymentBucketName = paramType;
+    cfnTemplate.Parameters.s3Key = paramType;
+    if (cfnTemplate.Resources.LambdaFunction.Type === 'AWS::Serverless::Function') {
+      cfnTemplate.Resources.LambdaFunction.Properties.CodeUri = {
         Bucket: Fn.Ref('deploymentBucketName'),
         Key: Fn.Ref('s3Key'),
       };
     } else {
-      cfnMeta.Resources.LambdaFunction.Properties.Code = {
+      cfnTemplate.Resources.LambdaFunction.Properties.Code = {
         S3Bucket: Fn.Ref('deploymentBucketName'),
         S3Key: Fn.Ref('s3Key'),
       };
     }
     storeS3BucketInfo(category, s3Bucket, envName, resourceName, s3Key);
   }
-  JSONUtilities.writeJson(cfnFilePath, cfnMeta);
+  JSONUtilities.writeJson(cfnFilePath, cfnTemplate);
```

Comment:
If the intent is to keep the template in it's original format (yaml or json), then use the `templateFormat` output from `readCFNTemplate` and pass it into the options of `writeCFNTemplate` here

Reply:
The current change just correctly read the template in Yaml or Json, but do save in Json only as per line 602 use the `JSONUtilities.writeJson()` 
But we can make a change to upload the template in the same format as the local file using the `writeCFNTemplate`.
I just went light in the changes to fix an issue that come to me in the cfn-flip project :-)